provider/openstack/cinder.go

```diff
@@ -250,16 +251,40 @@ func (s *cinderVolumeSource) createVolume(
 	// This helps to avoid a situation with all storage residing in a single
 	// AZ that upon failure would effectively take down attached instances
 	// whatever zone they were in.
+	// However, we first attempt to query the possible volume availability zones.
+	// If the API is old and does not support explicit volume AZs, or volumes can
+	// only be provisioned in say the default "nova" zone and the instance is in
+	// a different zone, we won't attempt to use the instance zone because that
+	// won't work and we'll get a 400 error back.
 	az := ""
 	if arg.Attachment != nil {
```

Comment:
it'd be nice if we could break this out into a different method.  make createVolume simpler to read.

az, err := s.availabilityZoneForVolume()

Reply:
good idea